libminifi/src/utils/RegexUtils.cpp

```diff
@@ -279,13 +372,39 @@ SMatch getLastRegexMatch(const std::string& string, const utils::Regex& regex) {
     current_str = search_result.suffix();
   }
 
-  auto diff = string.size() - last_match.string_.size();
-  last_match.string_ = string;
+  if (!last_match.ready()) {
+    return last_match;
+  }
+
+  struct MatchInfo {
+    bool matched;
+    size_t begin;
+    size_t end;
+  };
+
+  // we must save the sub matches' info in a way that does
+  // not get invalidated by SMatch::reset, and can be transferred
+  // to the updated match
+  std::vector<MatchInfo> match_infos;
+  match_infos.reserve(last_match.size());
   for (auto& match : last_match.matches_) {
-    if (match.match.rm_so >= 0) {
-      match.match.rm_so += diff;
-      match.match.rm_eo += diff;
-    }
+    match_infos.emplace_back(MatchInfo{
+      .matched = match.matched,
+      .begin = gsl::narrow<size_t>(std::distance(last_match.string_.cbegin(), match.first)),
+      .end = gsl::narrow<size_t>(std::distance(last_match.string_.cbegin(), match.second))
+    });
+  }
+  // offset of the start of the last match into the original string
+  auto offset = string.size() - last_match.string_.size();
+  last_match.reset(string);
+  last_match.ready_ = true;
+  for (auto& info : match_infos) {
+    size_t match_off = info.matched ? offset : 0;
+    last_match.matches_.emplace_back(SMatch::Regmatch{
```

Comment:
I would use push_back here for stronger type checking.

Reply:
@fgerlits noticed that clang-tidy advocated (or will advocate if the check is turned on) for using `emplace_back` instead of `push_back`